application/src/main/java/org/opentripplanner/graph_builder/module/DirectTransferGenerator.java

```diff
@@ -166,17 +176,29 @@ public void buildGraph() {
               continue;
             }
             // The TransferKey and PathTransfer are created differently for flex routing.
-            TransferKey transferKey = new TransferKey(sd.stop, stop, sd.edges);
-            PathTransfer pathTransfer = distinctTransfers.get(transferKey);
-            if (pathTransfer == null) {
-              // If the PathTransfer can't be found, it is created.
-              distinctTransfers.put(
-                transferKey,
-                new PathTransfer(sd.stop, stop, sd.distance, sd.edges, EnumSet.of(mode))
-              );
-            } else {
-              // If the PathTransfer is found, a new PathTransfer with the added mode is created.
-              distinctTransfers.put(transferKey, pathTransfer.withAddedMode(mode));
+            createPathTransfer(sd.stop, stop, sd, distinctTransfers, mode);
+          }
+        }
+        // Calculate transfers between stops that are visited by trips that allow cars, if configured.
```

Comment:
This method is getting too long. Please split it up into the various phases (default, flex, cars allowed) and then convert these line comments into javadoc.

Afterwards I'm satisfied and will approve.

Reply:
I split this into three functions